### ISSUE TITLE
Selecting a link in a WebView on iOS causes a NSInternalInconsistencyException

### ISSUE THREAD

**Issue by edgarssults** [dotnet/maui#17923]
### Description

Selecting any link in a WebView by pressing and holding always causes a crash on iOS.

`Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Completion handler passed to -[Microsoft_Maui_Platform_MauiWebViewUIDelegate webView:contextMenuConfigurationForElement:completionHandler:] was not called'`

There seems to be a problem with how the context menu is configured by MAUI. The problematic code is apparently [here](https://github.com/dotnet/maui/blob/9eb2fd370e1b5e14018fb64a4d6617871b8760c0/src/Core/src/Platform/iOS/MauiWebViewUIDelegate.cs#L20C24-L20C51).

It works fine on Android. Selecting normal text also works fine and the context menu appears.

Related issue: [https://github.com/dotnet/maui/issues/11544](https://github.com/dotnet/maui/issues/11544)

### Steps to Reproduce

1. Create a MAUI app with one WebView on the main page.
2. Load any URL that displays links (anchors) in the WebView.
3. Start the app on an iOS device.
4. Try to press and hold on the link to select it.

Actual behavior:
App crashes.

Expected behavior:
Link or part of the link is selected and a context menu appears.

### Link to public reproduction project repository

https://github.com/edgarssults/maui-ios-crash

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 17.0.3

### Did you find any workaround?

Creating a custom `WebViewHandler` that initializes a custom WebView on iOS where `AllowsLinkPreview` is set to false prevents this crash.

### Relevant log output

```shell
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Completion handler passed to -[Microsoft_Maui_Platform_MauiWebViewUIDelegate webView:contextMenuConfigurationForElement:completionHandler:] was not called'
```

**Comment by gerneio** [comment]
Having the an identical issue using the built-in MAUI WebView control on iOS (.NET 7). Found two workarounds:

1) Implement custom UIDelegate derived from MauiWebViewUIDelegate and make sure that SetContextMenuConfiguration:completionHandler() is always called (as demonstrated by [@anton-yashin here](https://github.com/dotnet/maui/issues/11544#issuecomment-1429831732)).

2) Set `WKWebView.AllowsLinkPreview` to `false` (as suggested by [OP @edgarssults here](https://github.com/dotnet/maui/issues/11544#issuecomment-1754487146)).

I had already implemented my own custom handler, so I implemented both fixes above, just in-case.
